sdk/batch.py

```diff
@@ -54,13 +53,18 @@ class Batch:
     sequence_builder: str
     start_datetime: Optional[str]
     end_datetime: Optional[str]
-    device_status: str = None
+    device_status: Optional[str] = None
     jobs: Dict[int, Job] = field(default_factory=dict)
     jobs_count: int = 0
     jobs_count_per_status: Dict[str, int] = field(default_factory=dict)
-    configuration: Optional[Dict] = None
+    configuration: Optional[Configuration] = None
+
+    def __post_init__(self) -> None:
+        self.configuration = Configuration.from_dict(self.configuration)
```

Comment:
This basically let's us keep the type as Configuration, but allows us to initialize with a dict.

Very happy to receive suggestions about this

Reply:
If `self.configuration` is a `Configuratioon` already,  what will this do?

Reply:
so I (confusingly) accept Configuration as an argument to`Configuration.from_dict` and it just returns the Configuration object.

Happy to get some ideas for how to do this better.

Reply:
I'd rather check it in post_init instead of inside the `from_dict` function but that's probably a matter of taste

Reply:
yeah, I did that in the first version, but I also thought it looked messy...
but it's also true that it's probably less confusing than accepting `None` and `Configuration` in a method called `from_dict`.

Reply:
went back on this